### ISSUE TITLE
Add a 'Post process now' button like Sickgear

### ISSUE THREAD

**Issue by ellnic** [barbequesauce/Watcher3#22]
Looks like the documentation indicates that this can be done, but there isn't a button in the Web UI. 

https://github.com/nosmokingbandit/watcher/wiki/Post-Processing#creating-a-post-processing-request

A button on the manage library or post processing page that triggers this would be ideal.

**Comment by barbequesauce** [comment]
That link you reference is for programmers to integrate other downloaders (airdcpp etc) into watcher, not for the directory scanner - which I think is what you're really after? 

If so, that's fairly easy - just need the clarification.

**Comment by ellnic** [comment]
Yes, forget that link - what I mean is a button that forces Watcher to check the download folder for items that it is responsible for post processing, then do that.

**Comment by scambra** [comment]
You can force postprocessing from settings > system
![image](https://user-images.githubusercontent.com/20515/66580466-bfee8f80-eb7e-11e9-853b-1ce58cea2585.png)
Are you asking something like that? Or UI to set folder to scan?

**Comment by ellnic** [comment]
Yeah, I didn't realise that was there either. 

You've hit the nail on the head @barbequesauce, it would be far better on the manage library page.

**Comment by barbequesauce** [comment]
Looking at this from a design perspective - a button on the manage library page makes sense. I'm struggling with *where*, though.

The row of buttons all take action on the movie list below the buttons. There's no place (currently) for anything unrelated to the list of movies.

I could create a row above the current row and add the one button - but that seems a waste of screen real estate?

Open to ideas here.

**Comment by ellnic** [comment]
What about in the Library drop down instead? Second to last above Stats...? It could link to a page with a [ Go! ] button, description and/or progress bar?

**Comment by barbequesauce** [comment]
Conceptually I like the idea of it being in the Library menu. I’ve been playing with it this evening and it led me to the discovery that the navbar is a stand-alone template snippet referred to in 12+ full-page templates. Fine and dandy, but the underlying js for the PP Scan button exists in the system module’s js.... 

TL;DR: I’m refactoring repetitive js code from various modules into one shared js that will be included on all pages. Codacy has been bitching about this anyway, may as well fixit.

**Comment by barbequesauce** [comment]
I keep breaking things doing this... @scambra can you take a whack at this please?

**Comment by scambra** [comment]
I have checked system.js, run a task calls execute_task, but I don't think we want to move it to shared.js, as it is, because it has assumptions about run button is in a table (e.g. render_task in row).

I have extracted more generic code to _execute_task to shared.js, and execute_task in system.js calls it. _execute_task still remove mdi-play-circle class, and re-adds it when it's done, although I don't know if menu item will have it.

@barbequesauce Will you add menu item to run PP task to menu? Or do you want me to do it?

**Comment by barbequesauce** [comment]
Yep I can get that. Did you commit shared.js?

**Comment by scambra** [comment]
Yes, I checked run tasks from system settings is not broken, and committed changes.

**Comment by barbequesauce** [comment]
Still struggling with this. The execute_task routine returns updates back to the element that calls it, so my menu looks like this before I click... 

<img width="224" alt="Screen Shot 2019-12-31 at 12 02 48 PM" src="https://user-images.githubusercontent.com/25070944/71628358-89487400-2bc5-11ea-9f5b-086681ea184b.png">

Then, after execution, it looks like this 🤣 

<img width="267" alt="Screen Shot 2019-12-31 at 12 00 28 PM" src="https://user-images.githubusercontent.com/25070944/71628324-6027e380-2bc5-11ea-8fa5-5dbefc53572e.png">

Other than including system.js in head.html, my only significant changes were adding the following snippet at line 36 of navbar.html. I'm sure it's just a case of changing out "this" as the element reference, but I can't figure out where to send the output? What's the js equivalent of /dev/null 😆 

```
                        <li>
	                        <a class="nav-link"> 
	                            <i class="mdi mdi-auto-fix" onclick="execute_task(event, this, 'PostProcessing Scan')"></i>${_('Manual Dir Scan')}
	                        </a>
                        </li>
                        <li>
```

**Comment by barbequesauce** [comment]
I'm thinking a new function using manual_task_execute from ajax.py might be the right answer. Thoughts?

**Comment by scambra** [comment]
I extracted code to run task to _execute_task funciton, and put it in shared.js, so system.js is not needed. execute_task in system.js calls _execute_task and replace tr with returned output, but _execute_task just run manual_task_execute from ajax.py, and changes classes (mdi-play-circle, mdi-circle, animate) when task starts and finishes.

Maybe _execute_task is not a good name, you could rename it, e.g. manual_execute_task, but you don't need to inlucde system.js in head.html

**Comment by barbequesauce** [comment]
Et voila!

**Comment by ellnic** [comment]
I bet you’re pleased that’s over :-)

**Comment by barbequesauce** [comment]
Learning experience ;)
